agent/agentcontainers/api_test.go

```diff
@@ -41,6 +45,103 @@ func (f *fakeDevcontainerCLI) Up(_ context.Context, _, _ string, _ ...agentconta
 	return f.id, f.err
 }
 
+// fakeWatcher implements the watcher.Watcher interface for testing.
+// It allows controlling what events are sent and when.
+type fakeWatcher struct {
+	t           testing.TB
+	events      chan *fsnotify.Event
+	closeNotify chan struct{}
+	addedPaths  []string
+	closed      bool
+	nextCalled  chan struct{}
+	nextErr     error
+	closeErr    error
+}
```

Comment:
suggestion: move into `watcher/watchertest` package and export

Reply:
I don't want to increase package surface, I'd like to see at least a few more uses before trying to generalize into a useful test package.

Reply:
Fair!